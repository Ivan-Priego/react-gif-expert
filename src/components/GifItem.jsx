import PropTypes from 'prop-types'

export const GifItem = ( {title,url,id} ) => {
  return (
    <li className="card">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </li>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}

// GifItem.defaultProps = {
//   title : "Hola no hay titulo",
//   url : "Hola no hay URL",
// }