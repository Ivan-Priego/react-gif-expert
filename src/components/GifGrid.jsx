import { PropTypes } from "prop-types"
import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"

export const GifGrid = ({category}) => {
  const { images, isLoading } = useFetchGifs ( category )
   return (
      <>
        <h3> {category} </h3>
        {
          isLoading && ( <h2> Cargando ...</h2> ) // esto es un and logico una condicion enuna sola linea
        }
        {/*<h5>{counter}</h5>
        <button onClick={ () => setCounter(counter +1 ) } >+1</button> */}
        <div className="card-grid">
          {
            images.map( image => (
                < GifItem  
                  key={ image.id }
                  { ...image }
                />
              ))
          }
        </div>
      </>
  )
}
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}