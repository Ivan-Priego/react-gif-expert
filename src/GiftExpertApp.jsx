import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GiftExpertApp = () => {
  const [ categories, setCategories ] = useState(['One Punch'])
  const onAddCategory = ( newCategory ) => {
    if (categories.includes(newCategory)) return;
    //console.log(newCategory)
    setCategories ( [  newCategory, ...categories] )
  }
  return (
    <>
      {/* <p>https://es.legacy.reactjs.org/docs/faq-structure.html</p> */}
      {/* totulo */}
      <h1>GiftExpertApp</h1>
      {/* input */}
      < AddCategory 
        //setCategories={setCategories} 
        onNewCategory = { onAddCategory }
      />
      {/* listado */}
        {/* <ol> */}
          {
            categories.map( (category) => (
              <GifGrid 
                key={category}
                category={category}
              />
            ))
          }
        {/* </ol> */}
        {/* gif Item */}
    </>    
  )
}
