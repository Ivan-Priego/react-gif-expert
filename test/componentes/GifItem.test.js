import { render, screen } from "@testing-library/react" 
import { GifItem } from "../../src/components/GifItem";
import { GiftExpertApp } from "../../src/GiftExpertApp";

describe('Purebas en GifItem.js', () => { 

  const title='Titulo test'
  const url ='http://urltest.mx/img.jpg'

  test('Debe tomar un snapshot del componente', () => { 
    
    const { container } = render ( < GifItem title={title} url={url} /> )
    expect (container).toMatchSnapshot();

   })

   test('debe mostara la imagen con el url y alt indicado ', () => { 
    
      render ( < GifItem title={title} url={url} /> )
      //screen.debug();
      //console.log(screen.getByRole('img').src)
      //console.log(screen.getByRole('img').alt)
      //expect (screen.getByRole('img').src).toBe( url )
      //expect (screen.getByRole('img').alt).toBe( title )

      const { src , alt } = screen.getByRole('img')
      expect( src ).toBe( url )
      expect( alt ).toBe( title )

    })

    test('debe mostara el titulo en el componente', () => { 
      render ( < GifItem title={title} url={url} /> )

      expect( screen.getByText(title)).toBeTruthy();
    })
 })