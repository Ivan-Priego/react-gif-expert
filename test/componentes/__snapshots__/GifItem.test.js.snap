// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purebas en GifItem.js Debe tomar un snapshot del componente 1`] = `
<div>
  <li
    class="card"
  >
    <img
      alt="Titulo test"
      src="http://urltest.mx/img.jpg"
    />
    <p>
      Titulo test
    </p>
  </li>
</div>
`;
